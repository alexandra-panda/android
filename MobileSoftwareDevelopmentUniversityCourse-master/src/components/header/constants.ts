export default {
  defaultHeaderHeight: 50,
  defaultHeaderButtonHeight: 30,
}
