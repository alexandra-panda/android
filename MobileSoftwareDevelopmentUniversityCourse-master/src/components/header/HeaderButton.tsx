import React, { FC, ReactElement } from 'react'
import { StyleSheet, View, Text, TouchableOpacity, GestureResponderEvent } from 'react-native'

import constants from '@/components/header/constants'

type HeaderButtonPropType = {
  title: string
  onClickHandler: (e: GestureResponderEvent, routeName: string) => void
  routeDir: string
}

export const HeaderButton: FC<HeaderButtonPropType> = (props): ReactElement => {
  return (
    <TouchableOpacity
      onPress={(e) => props.onClickHandler(e, props.routeDir)}
      activeOpacity={0.7}
      style={styles.headerButton}
    >
      <View>
        <Text style={styles.customBtnText}>{props.title}</Text>
      </View>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  headerButton: {
    backgroundColor: '#7a9ea3',
    paddingHorizontal: 7,
    paddingVertical: 3,
    marginHorizontal: 5,
    height: constants.defaultHeaderButtonHeight,
    borderRadius: 2,
    maxWidth: 90,
    minWidth: 70,
    justifyContent: 'center',
    alignItems: 'center',
    transform: [{ translateY: -constants.defaultHeaderButtonHeight / 2 }],
    top: constants.defaultHeaderHeight / 2,
  },
  customBtnText: {
    fontSize: 13,
    fontWeight: 'bold',
    color: '#fff',
  },
})
