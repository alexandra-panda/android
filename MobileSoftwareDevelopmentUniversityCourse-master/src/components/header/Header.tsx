import React, { FC, ReactElement } from 'react'
import { StatusBar, Dimensions, StyleSheet, ScrollView, GestureResponderEvent } from 'react-native'
import { StackHeaderProps } from '@react-navigation/stack'

import { HeaderButton } from '@/components/header/HeaderButton'

import { routes } from '@/routes/routes'

import constants from '@/components/header/constants'

type HeaderPropsType = StackHeaderProps

export const Header: FC<HeaderPropsType> = ({ navigation }): ReactElement => {
  const buttonOnClickFunc = function (e: GestureResponderEvent, routeName: string) {
    navigation.navigate(routeName)
  }

  return (
    <ScrollView
      horizontal={true}
      style={styles.scrollWrapper}
      contentContainerStyle={styles.elementsContainer}
    >
      <StatusBar hidden={true} />

      {Object.keys(routes)
        .filter((key) => routes[key].isHeaderButton)
        .map((key) => (
          <HeaderButton
            key={key}
            routeDir={routes[key].routeName}
            title={routes[key].headerButtonName as string}
            onClickHandler={buttonOnClickFunc}
          />
        ))}
    </ScrollView>
  )
}

const styles = StyleSheet.create({
  scrollWrapper: {
    width: Dimensions.get('window').width,
    height: constants.defaultHeaderHeight,
    backgroundColor: '#256E7B',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  elementsContainer: {
    backgroundColor: 'transparent',
    height: constants.defaultHeaderHeight,
  },
})
