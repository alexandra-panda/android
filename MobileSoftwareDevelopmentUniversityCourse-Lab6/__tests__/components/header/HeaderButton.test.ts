/**
 * @jest-environment jsdom
 */
/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-empty-function */
import 'react-native'
import { TouchableOpacity } from 'react-native'
import { createElement } from 'react'
import { shallow } from 'enzyme'

import { HeaderButton, HeaderButtonPropType } from '@/components/header/HeaderButton'

describe('Header Button test case', () => {
  let headerButtonProps: HeaderButtonPropType

  beforeEach(() => {
    headerButtonProps = {
      onClickHandler: () => {},
      title: 'tITle_314134',
      routeDir: 'HomeAboutRouteExample341341',
    }
  })

  test('Header Button renders correctly', () => {
    const component = shallow(createElement(HeaderButton, headerButtonProps, null))
    expect(component).toMatchSnapshot()
  })

  test('Click event invoke callback', () => {
    const onClickHandlerMock = jest.fn()
    headerButtonProps.onClickHandler = onClickHandlerMock

    const component = shallow(createElement(HeaderButton, headerButtonProps, null))
    const btnProps = component.find(TouchableOpacity).props() as any

    btnProps.onPress({})
    btnProps.onPress({})
    btnProps.onPress({})

    expect(onClickHandlerMock).toHaveBeenCalledTimes(3)
  })
})
