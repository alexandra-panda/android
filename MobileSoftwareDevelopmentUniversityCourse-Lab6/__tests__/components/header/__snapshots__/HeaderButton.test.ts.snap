// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Button test case Header Button renders correctly 1`] = `
<ForwardRef
  activeOpacity={0.7}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7a9ea3",
      "borderRadius": 2,
      "flex": 0,
      "justifyContent": "center",
      "marginHorizontal": 5,
      "marginVertical": 4,
      "maxWidth": 90,
      "minWidth": 70,
      "paddingHorizontal": 7,
      "paddingVertical": 3,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 13,
        }
      }
    >
      tITle_314134
    </Text>
  </View>
</ForwardRef>
`;
